endpoints/cookie_sync_test.go

```diff
@@ -2207,3 +2268,129 @@ func getDefaultActivityConfig(componentName string, allow bool) *config.AccountP
 		},
 	}
 }
+
+func TestSetCookieDeprecationHeader(t *testing.T) {
+	getTestRequest := func(addCookie bool) *http.Request {
+		r := httptest.NewRequest("POST", "/cookie_sync", nil)
+		if addCookie {
+			r.AddCookie(&http.Cookie{Name: receiveCookieDeprecation, Value: "1"})
+		}
+		return r
+	}
+
+	tests := []struct {
+		name                            string
+		responseWriter                  http.ResponseWriter
+		request                         *http.Request
+		account                         *config.Account
+		expectedCookieDeprecationHeader bool
+	}{
+		{
+			name:                            "not-present-account-nil",
+			request:                         getTestRequest(false),
+			responseWriter:                  httptest.NewRecorder(),
+			expectedCookieDeprecationHeader: false,
```

Comment:
Please specifically add `account: nil` to make the test setup clear. Same for the "present-account-nil" case.